### DISCUSSION TITLE
Feature request: Import svg files

### DISCUSSION THREAD

**Discussion by shimonshomron** [keiwando/toonsquid#70] [Ideas]
Currently the app allow for importing raster images. However since it has vector drawing capabilities, its a miss that I can't import ready made svg files created on more vector-oriented apps/softwares/assets

**Comment by JalilOz** [comment]
I would suggest also the feature to export to SVG.

**Comment by Kosdaos1** [comment]
+1 I need animation frames in vector format for my work, if it were possible to do it in toonsquid, it would be great!

**Comment by arcdiablo** [comment]
Would be amazing , or better tools to design

**Comment by Kosdaos1** [comment]
Yes, it's a pity that now there is no function to add vector drawings from other applications.

**Comment by cirquesbreeze** [comment]
I also wanted to bring up the suggestion that it would be fantastic if ToonSquid could support the import of SVG files. This addition would greatly benefit users like myself who work with vector-based assets made outside the app (for example, Illustrator).  

SVG import would enable seamless integration of existing vector artwork, providing the advantage of infinite scalability and preserving sharpness across different resolutions.

I believe that implementing SVG import functionality in ToonSquid would be a valuable addition and further elevate the app’s capabilities. Thank you for considering this feature request, and I look forward to any future updates that may include SVG support.

**Comment by arcdiablo** [comment]
Would be amazing to import svg, now I’m turning layers to png files , but if I want to animate something have to plan it before exporting to png , and importing to ToonSquid, in a vector image would be more direct and versatile to do it , more like old flash

**Reply by Kosdaos1** [reply]
Right!

**Comment by ruvaen** [comment]
imporring svg is a must and would be amazing for this app!

**Comment by desing-barce** [comment]
It would be very interesting to be able to import SVG, the difference for the design would be spectacular and even more so now that the Procreate competition is approaching. The difference always wins!!!

**Comment by budaloops** [comment]
It’s faster to create vector drawings in specialized vector drawing apps, such as Affinity Designer and then import it into ToonSquid, instead of using Toonsquid’s vector drawing tools at its current state.

I’m happy the app even has drawing tools because it’s the first animation app to do so. However it doesn’t compare to other vector drawing apps for professional use.

I know it takes a lot of work to improve the drawing tools, and there are already good drawing apps out there.

To add the import SVG feature would make ToonSquid even more specialized. As of now it will be the only app that allows vector animation in a professional workflow where there are tight schedules. And the go-to choice for vector animation for those of us who use an iPad as our main device.

To be able to import vector assets into ToonSquid will save me so much time when animating client work. Right now it would take too much time to recreate all my vector assets inside of ToonSquid, but as soon as “import SVG” becomes available then it’s a no-brainer to add ToonSquid to my workflow as a freelance graphic designer.

**Reply by kevincreative** [reply]
Currently my workflow is exporting pngs out of Designer and it works pretty good but I think an svg workflow might be lighter and of course be a lot more scalable. I have never used any of the drawing features of TS as I'm much more comfortable and faster in Designer on the desktop.

**Comment by SlamRockwell** [comment]
Man I hoped that I was gonna be the first person to post about this but a year and 3 months later and this still isn't a feature. Drawing in this app is a nightmare honestly. 

Without even spending time creating new features, just allowing artists to be able to batch import SVG files and name them would make it not only one of the best animation apps on the iPad but one of the best animation apps period. People would be using it for student films and personal projects everywhere. The potential for a whole new generation of people now creating vector animations on their tablet. 

I wanted to use it to make my next film but sadly it's unusable for more advanced animation work and I'm probably just going to have to use my Adobe Animate crack

**Comment by arcdiablo** [comment]
There is a new animation app, “ linearity move” , that allow SVG to be imported, it’s a  feature for the workflow that we where asking to toonsquid to implement, I love the app , it’s logic and structure, similar to flash or animate  but without this feature is to complicate to work with it

**Reply by budaloops** [reply]
I'm loving the Linearity Move workflow. It's still a young app but the SVG import feature is a big one. Here is an animation I made with it: 

https://drive.google.com/file/d/1EqBWHT4ituysFrGKXU-opl3lQDGr9bhk/view?usp=drivesdk

I was able to import this Lighthouse and Ship illustration and animate it

**Comment by alex-dearden** [comment]
> Currently the app allow for importing raster images. However since it has vector drawing capabilities, its a miss that I can't import ready made svg files created on more vector-oriented apps/softwares/assets

+1 for SVG import!

**Comment by All123channel-cell** [comment]
Feature Request: Live Lasso Closing Preview

When using the freehand lasso tool, please add a temporary preview line from the current Apple Pencil position back to the starting point (similar to ibis Paint and Photoshop). This makes it much easier to judge where the selection will close before lifting the pencil.
